Projects/Server/TileMatrix/TileMatrix.cs

```diff
@@ -434,9 +439,45 @@ private unsafe LandTile[] ReadLandBlock(int x, int y)
         }
     }
 
+    public long FindOffset(long offset)
+    {
+        var total = 0;
+
+        for (var i = 0; i < _uopMapEntries.Length; ++i)
+        {
+            var entry = _uopMapEntries[i];
+            var newTotal = total + entry.Size;
+
+            if (offset < newTotal)
+            {
+                return entry.Offset + (offset - total);
+            }
+
+            total = newTotal;
+        }
+
+        return -1;
+    }
+
+    private void ConvertToMapEntries(FileStream stream)
+    {
+        // Sorting by offset to make seeking faster
+        Array.Sort(_uopMapEntries, (a, b) => a.Offset.CompareTo(b.Offset));
```

Comment:
This is only doable because I don't think the hashed file name order is important. I also don't think those files match the order field in the data block.